src/movies.js

```diff
@@ -1,28 +1,138 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const directors = movies.map((movie) => movie.director);
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(bestDirector) {
+  return bestDirector.filter(
+    (stevenSpielberg) => 
+      stevenSpielberg.director === "Steven Spielberg" && stevenSpielberg.genre.includes("Drama")).length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(averageMovies) {
+  if (!averageMovies.length){
+    return 0;
+  }
+
+  let total = averageMovies.reduce((a, b) => {
+    if (b.rate) {
+      return a + b.rate;
+    } else {
+      return a;
+    }
+  }, 0);
+  return Number((total / averageMovies.length).toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(dramaMovies) {
+  let moviesInspector = dramaMovies.filter((theMovie) =>
+    theMovie.genre.includes("Drama")
+  );
+  return scoresAverage(moviesInspector);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(scoresAverage) {
+  let arrayOrdered = [];
```

Comment:
aquí no estáis usando el array que entra como parámetro. Podemos hacernos una copia en arrayOrdered de esta forma:
`let arrayOrdered = [...scoresAverage];`